boot/zephyr/CMakeLists.txt

```diff
@@ -365,3 +365,9 @@ zephyr_library_sources(
   ${BOOT_DIR}/zephyr/nrf_cleanup.c
 )
 endif()
+
+if(CONFIG_MCUBOOT_HW_DOWNGRADE_PREVENTION)
+zephyr_library_sources(
+  ${BOOT_DIR}/zephyr/nrf_nv_counters.c
+)
+endif()
```

Comment:
We can avoid some forking if we move nrf_nv_counters.c into nrf and also move this CMake code into nrf.

Reply:
and also reduce the amount of manifest PRs we need to deal with.